src/OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -218,48 +219,44 @@ class SubmissionTracker : public VulkanLayerProducer::CaptureStatusListener {
     // submission. We will not write timestamps in this case and thus don't store any information
     // other than the debug markers then.
     {
-      absl::WriterMutexLock lock(&mutex_);
       CHECK(!command_buffer_to_state_.contains(command_buffer));
       command_buffer_to_state_[command_buffer] = {};
     }
-    if (!IsCapturing()) {
+    if (!is_capturing_) {
       return;
     }
 
     uint32_t slot_index = RecordTimestamp(command_buffer, VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT);
     {
-      absl::WriterMutexLock lock(&mutex_);
       CHECK(command_buffer_to_state_.contains(command_buffer));
       command_buffer_to_state_.at(command_buffer).command_buffer_begin_slot_index =
           std::make_optional(slot_index);
     }
   }
 
   void MarkCommandBufferEnd(VkCommandBuffer command_buffer) {
-    if (!IsCapturing()) {
+    absl::WriterMutexLock lock(&mutex_);
```

Comment:
Just to mention that with the argument below, a read lock would be fine, but agree to make it simpler to reason about.

Reply:
This is writing command_buffer_end_slot_index, which may also be written / reset by OnCaptureFinished at the same time. Or am I missing something?

Reply:
Thanks. I was missing this.